crates/ruma-common/src/thirdparty.rs

```diff
@@ -37,16 +37,30 @@ pub struct Protocol {
     pub field_types: BTreeMap<String, FieldType>,
 
     /// A list of objects representing independent instances of configuration.
-    pub instances: Vec<ProtocolInstance>,
+    pub instances: Vec<I>,
 }
 
-/// Initial set of fields of `Protocol`.
+impl<I> Protocol<I> {
+    /// Convert this `Protocol<I>` to a `Protocol<J>`.
+    pub fn into<J: From<I>>(self) -> Protocol<J> {
```

Comment:
Why as an inherent method, rather than a `From` impl?

Reply:
If I use this code:

```rust
impl<I, J> From<Protocol<I>> for Protocol<J>
where
  J: From<I>
{ .. }
```

It conflicts with the blanket implementation of `From<Protocol<I>> for Protocol<I>`.

Reply:
Ah, right. Not a big fan of "overloading" name like this he but I guess it's fine.